contracts/helpers/TestReentrantToken.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: SEE LICENSE IN LICENSE
+pragma solidity 0.8.19;
+
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import { ICarbonController } from "../carbon/interfaces/ICarbonController.sol";
+import { Order, TradeAction } from "../carbon/Strategies.sol";
+import { Token } from "../token/Token.sol";
+
+/**
+ * @dev token contract which attempts to re-enter CarbonController
+ */
+contract TestReentrantToken is ERC20 {
+    ICarbonController private immutable _carbonController;
+
+    enum ReenterFunctions {
+        CREATE_PAIR,
+        CREATE_STRATEGY,
+        UPDATE_STRATEGY,
+        DELETE_STRATEGY,
+        TRADE_BY_SOURCE_AMOUNT,
+        TRADE_BY_TARGET_AMOUNT,
+        WITHDRAW_FEES
+    }
+
+    // which function to reenter using transferFrom
+    uint8 private _reenterFunctionIndex;
```

Comment:
Can you define it as `ReenterFunctions`? This way you won't need the casts below.

Reply:
It's defined as uint8 so as to be able to pass it as a fuzzing parameter in the test testStrategyCreationRevertsIfReentrancyIsAttempted(uint8 reenterFunctionIndex) in Strategies.t.sol - so that we don't need to define several tests for it.

Reply:
fixed